crates/editor/src/hunk_diff.rs

```diff
@@ -542,12 +543,12 @@ fn create_diff_base_buffer(buffer: &Model<Buffer>, cx: &mut AppContext) -> Optio
     buffer
         .update(cx, |buffer, _| {
             let language = buffer.language().cloned();
-            let diff_base = buffer.diff_base().map(|s| s.to_owned());
-            Some((diff_base?, language))
+            let diff_base = buffer.diff_base()?.clone();
+            Some((buffer.line_ending(), diff_base, language))
         })
-        .map(|(diff_base, language)| {
+        .map(|(line_ending, diff_base, language)| {
             cx.new_model(|cx| {
-                let buffer = Buffer::local(diff_base, cx);
+                let buffer = Buffer::local_normalized(diff_base, line_ending, cx);
```

Comment:
Before, Buffer creation allowed strings only, so that it gets normalized, its line ending will be detected, etc.
I had to create another constructor that skips the normalization and accepts whatever ropes passed into that: for diff base case, that normalization step should not be applied, so seems like a legit change.